src/server/src/main/java/io/cassandrareaper/storage/cluster/MemClusterDao.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2014-2017 Spotify AB
+ * Copyright 2016-2019 The Last Pickle Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.storage.cluster;
+
+import io.cassandrareaper.core.Cluster;
+import io.cassandrareaper.storage.events.MemEventsDao;
+import io.cassandrareaper.storage.repairrun.MemRepairRunDao;
+import io.cassandrareaper.storage.repairschedule.MemRepairScheduleDao;
+import io.cassandrareaper.storage.repairunit.MemRepairUnitDao;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.StringUtils;
+
+public class MemClusterDao implements ICluster {
+  public final ConcurrentMap<String, Cluster> clusters = Maps.newConcurrentMap();
+  private final MemRepairUnitDao memRepairUnitDao;
+  private final MemRepairRunDao memRepairRunDao;
+  private final MemRepairScheduleDao memRepairScheduleDao;
+
+  private final MemEventsDao memEventsDao;
+
+  public MemClusterDao(MemRepairUnitDao memRepairUnitDao,
+                       MemRepairRunDao memRepairRunDao,
+                       MemRepairScheduleDao memRepairScheduleDao,
+                       MemEventsDao memEventsDao) {
+    this.memRepairUnitDao = memRepairUnitDao;
+    this.memRepairRunDao = memRepairRunDao;
+    this.memRepairScheduleDao = memRepairScheduleDao;
+    this.memEventsDao = memEventsDao;
+  }
+
+  @Override
+  public Collection<Cluster> getClusters() {
+    return clusters.values();
+  }
+
+  @Override
+  public boolean addCluster(Cluster cluster) {
+    assert addClusterAssertions(cluster);
+    Cluster existing = clusters.put(cluster.getName(), cluster);
+    return existing == null;
+  }
+
+  @Override
+  public boolean updateCluster(Cluster newCluster) {
+    addCluster(newCluster);
+    return true;
+  }
+
+  public boolean addClusterAssertions(Cluster cluster) {
+    Preconditions.checkState(
+          Cluster.State.UNKNOWN != cluster.getState(),
+          "Cluster should not be persisted with UNKNOWN state");
+
+    // TODO – unit tests need to also always set the paritioner
+    //Preconditions.checkState(cluster.getPartitioner().isPresent(), "Cannot store cluster with no partitioner.");
```

Comment:
**Suggestion:** Maybe we could delete these lines if they were carried over from the existing code?

```suggestion
```

Reply:
Done!